community/kernel/src/main/java/org/neo4j/kernel/impl/store/StoreType.java

```diff
@@ -221,21 +222,21 @@ void close( NeoStores me, Object object )
     /**
      * Determine type of a store base on a store file name.
      *
-     * @param storeFileName - name of the store to map
-     * @return store type of specified file
-     * @throws IllegalStateException if can't determine store type for specified file
+     * @param fileName - name of the store to map
+     * @return an {@link Optional} that wraps the matching store type of the specified file,
+     * or {@link Optional#empty()} if the given file name does not match any store file name.
      */
-    public static StoreType typeOf( String storeFileName )
+    public static Optional<StoreType> typeOf( String fileName )
     {
         StoreType[] values = StoreType.values();
         for ( StoreType value : values )
         {
-            if ( value.getStoreName().equals( storeFileName ) ||
-                    storeFileName.equals( MetaDataStore.DEFAULT_NAME + value.getStoreName() ) )
+            if ( value.getStoreName().equals( fileName ) ||
```

Comment:
To match both (w/ and w/o `neostore.` in front) looks like a source of confusion and a potential edge-case problems finding unrelated files. It'd be better to decide exactly what it looks for, document it and have all callers adhere to it.